provisioning/service/src/Config/TwinState.cs

```diff
@@ -0,0 +1,164 @@
+    ///                 "$lastUpdated":"2017-09-21T02:07:44.238Z",
+    ///                 "$lastUpdatedVersion":4,
+    ///                 "SpeedUnity":{
+    ///                     "$lastUpdated":"2017-09-21T02:07:44.238Z",
+    ///                     "$lastUpdatedVersion":4
+    ///                 }
+    ///             },
+    ///             "$version":4
+    ///         }
+    ///         "properties":{
+    ///             "desired": {
+    ///                 "MaxSpeed":{
+    ///                     "Value":500,
+    ///                     "NewValue":300
+    ///                 },
+    ///                 "$metadata":{
+    ///                     "$lastUpdated":"2017-09-21T02:07:44.238Z",
+    ///                     "$lastUpdatedVersion":4,
+    ///                     "MaxSpeed":{
+    ///                         "$lastUpdated":"2017-09-21T02:07:44.238Z",
+    ///                         "$lastUpdatedVersion":4,
+    ///                         "Value":{
+    ///                             "$lastUpdated":"2017-09-21T02:07:44.238Z",
+    ///                             "$lastUpdatedVersion":4
+    ///                         },
+    ///                         "NewValue":{
+    ///                             "$lastUpdated":"2017-09-21T02:07:44.238Z",
+    ///                             "$lastUpdatedVersion":4
+    ///                         }
+    ///                     }
+    ///                 },
+    ///                 "$version":4
+    ///             }
+    ///         }
+    ///     }
+    /// }
+    /// </code>
+    /// </example>
+    /// <seealso cref="https://docs.microsoft.com/en-us/rest/api/iot-dps/deviceenrollment">Device Enrollment</seealso>
+    /// <seealso cref="https://docs.microsoft.com/en-us/rest/api/iot-dps/deviceenrollmentgroup">Device Enrollment Group</seealso>
+    /// <seealso cref="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-device-twins">Understand and use device twins in IoT Hub</seealso>
+    /// <seealso cref="https://docs.microsoft.com/en-us/rest/api/iothub/devicetwinapi">Device Twin Api</seealso>
+    public class TwinState
+    {
+        /// <summary>
+        /// Getter and setter the <see cref="TwinState"/> tags.
+        /// </summary>
+        [JsonProperty(PropertyName = "tags", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public TwinCollection Tags { get; set; }
+
+        [JsonProperty(PropertyName = "properties", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        private TwinProperties _properties;
+
+        /// <summary>
+        /// Getter and setter the <see cref="TwinState"/> properties.
+        /// </summary>
+        [JsonIgnore]
+        public TwinCollection DesiredProperties
+        {
+            get
+            {
+                /* SRS_TWIN_STATE_21_002: [If the _properties is null, the get.DesiredProperties shall return null.] */
+                if (_properties == null)
+                    return null;
```

Comment:
Please always use braces or write the if statement on a single line.

Reply:
This is also a behavior breaking change: it would never return null before.

Reply:
Preview behavior was wrong and under `preview` label. So, we can fix it now.

Reply:
Braces done